server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -87,6 +90,9 @@ public void preProcess(SearchContext context) {
 
     @Override
     public void execute(SearchContext context) {
+
+        LOGGER.trace("{}", new SearchContextRequestLog(context));
```

Comment:
It might make sense to guard this log statement? (i.e. `if LOGGER.isTraceEnabled())`?)

Reply:
++, this will avoid the creation of a new object on each query